### ISSUE TITLE
Publish-CRL throws: An attempt was made to access an uninitialized object

### ISSUE THREAD

**Issue by c3rberus** [PKISolutions/PSPKI#163]
Using PSPKI 3.7.2, when trying to run Publish-CRL -CertificationAuthority XXXXIntermediateCAv2 I get the following error:

```
Exception calling "PublishCRL" with "2" argument(s): "An attempt was made to access an uninitialized object."
At C:\Program Files\WindowsPowerShell\Modules\PSPKI\3.7.2\Server\Publish-CRL.ps1:15 char:4
+             $CA.PublishCRL($DeltaOnly, $UpdateFile)
+             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], MethodInvocationException
    + FullyQualifiedErrorId : UninitializedObjectException
```

If I try this another method, I get this..

```
PS C:\Windows\system32> Get-CertificationAuthority | Publish-CRL
Exception calling "PublishCRL" with "2" argument(s): "CCertAdmin::PublishCRLs: The parameter is incorrect. 0x80070057
(WIN32: 87 ERROR_INVALID_PARAMETER)"
At C:\Program Files\WindowsPowerShell\Modules\PSPKI\3.7.2\Server\Publish-CRL.ps1:15 char:4
+             $CA.PublishCRL($DeltaOnly, $UpdateFile)
+             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], MethodInvocationException
    + FullyQualifiedErrorId : ArgumentException
```

Any idea why this is happening?

**Comment by Crypt32** [comment]
do you receive your CAs when calling `Get-CertificationAuthority`?

**Comment by kempd** [comment]
Same issue here, also on v3.7.2 
Get-CertificationAuthority results in:
```
Name                 : <redacted>
DisplayName          : <redacted>
ComputerName         : <redacted>
ConfigString         : <redacted>
DistinguishedName    : CN=<redacted>,CN=Enrollment Services,CN=Public Key 
                       Services,CN=Services,CN=Configuration,<redacted>
Type                 : Enterprise Subordinate CA
IsEnterprise         : True
IsRoot               : False
OperatingSystem      : Microsoft Windows Server 2019 Datacenter
IsAccessible         : True
RegistryOnline       : True
ServiceStatus        : Running
SetupStatus          : ServerInstall, SecurityUpgraded, ServerIsUptoDate
Certificate          : [Subject]
                         <redacted>
                       
                       [Issuer]
                         <redacted>
                       
                       [Serial Number]
                         <redacted>
                       
                       [Not Before]
                         <redacted>
                       
                       [Not After]
                        <redacted>
                       
                       [Thumbprint]
                         <redacted>
                       
BaseCRL              : 
DeltaCRL             : 
EnrollmentServiceURI : 
EnrollmentEndpoints  : {}
```

Publish-CRL -CertificationAuthority $ca.Name gives the same error as OP.

**Comment by jimbju** [comment]
Getting the same error as OP:
```powershell
$CAName = 'My CA'
$CAObj = Get-CertificationAuthority -Name $CAName
$CAObj | Publish-CRL
```

```
Exception calling "PublishCRL" with "2" argument(s): "CCertAdmin::PublishCRLs: The parameter is incorrect. 0x80070057 (WIN32: 87 ERROR_INVALID_PARAMETER)"
At C:\Program Files\PKI Solutions\PowerShell Modules\PSPKI\Server\Publish-CRL.ps1:15 char:4
+             $CA.PublishCRL($DeltaOnly, $UpdateFile)
+             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], MethodInvocationException
    + FullyQualifiedErrorId : ArgumentException
```

**Comment by Crypt32** [comment]
@jimbju what returns your `$CAObj` variable?

**Comment by jimbju** [comment]
@Crypt32 contents of `$CAObj`:
```powershell
DisplayName             ComputerName              IsAccessible ServiceStatus Type                     
-----------                  ------------                       ------------ ------------- ----                     
My CA                       mycacomputername...      True         Running       Enterprise Subordinate CA
```

**Comment by Crypt32** [comment]
Fixed in [v4.0.0](https://github.com/PKISolutions/PSPKI/releases/tag/v4.0.0)
